compiler/resolution/lowerIterators.cpp

```diff
@@ -2523,27 +2523,32 @@ expandForLoop(ForLoop* forLoop) {
         break;
       }
 
-      if (isBoundedIterator(iterFn)) {
-        if (testBlock == NULL) {
-          if (isNotDynIter) {
-            // note that we have found the first test
-            testBlock = buildIteratorCall(NULL, HASMORE, iterators.v[i], children);
-
-          } else {
-            // note that we have found the first test block and add checks for
-            // more before and at the end of the loop. As mentioned above,
-            // dynamic iterators generate things that can't be in the header of
-            // the c for loop, so we generate a simple bool variable to put at
-            // the test of the c for loop, and update that condition var before
-            // the loop is run, and at the end of each iteration.
-            forLoop->insertBefore(new DefExpr(cond));
-            forLoop->insertBefore(buildIteratorCall(cond, HASMORE, iterators.v[i], children));
-            forLoop->insertAtTail(buildIteratorCall(cond, HASMORE, iterators.v[i], children));
-
-            testBlock = new BlockStmt(new SymExpr(cond));
-          }
+      // If we haven't yet generated a test to terminate the loop and
+      // are in a zippered context, we'll set it up based on this
+      // iterator which is presumably the first.
+      if (testBlock == NULL) {
+        if (!isBoundedIterator(iterFn) && iterators.n > 1) {
+          USR_WARN(forLoop, "The behavior of zippered serial loops whose leader is an unbounded range has changed in this release; to maintain the previous behavior, swap a bounded iterand into the leader position (the first expression in the 'zip(...)').");
```

Comment:
How about we point them to #20051 ? The warning is lacking an explanation of what the change is.

Reply:
I'm not a huge fan of leaning too much on issue numbers in error messages (though I know we do it sometimes) for fear of someone not having internet access.  Good point about the lack of an explanation though.  I'm updating to:

> The behavior of zippered serial loops driven by unbounded ranges has been fixed in this release to act as though they were conceptually infinite; to maintain the previous behavior, swap a bounded iterand into the first expression of the 'zip(...)'.

if that sounds OK to you.